engine/primitive/src/main/java/io/deephaven/engine/primitive/value/iterator/ValueIteratorOfChar.java

```diff
@@ -0,0 +1,159 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.primitive.value.iterator;
+
+import io.deephaven.engine.primitive.function.CharToIntFunction;
+import io.deephaven.engine.primitive.iterator.CloseablePrimitiveIteratorOfChar;
+import io.deephaven.util.QueryConstants;
+import io.deephaven.util.annotations.FinalDefault;
+import io.deephaven.util.type.TypeUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.PrimitiveIterator;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public interface ValueIteratorOfChar extends CloseablePrimitiveIteratorOfChar, ValueIterator<Character> {
+
+    @Override
+    @FinalDefault
+    default Character next() {
+        return TypeUtils.box(nextChar());
+    }
+
+    @Override
+    @FinalDefault
+    default void forEachRemaining(@NotNull final Consumer<? super Character> action) {
+        forEachRemaining((final char element) -> action.accept(TypeUtils.box(element)));
```

Comment:
This loses an optimization vs the superinterface. Maybe not a very valid or important one? But if you have something that implements `Consumer` and `CharConsumer`, but naively invoke the `Iterator` version of this, you'll have a worse time than you might choose.